synapse/http/matrixfederationclient.py

```diff
@@ -188,6 +188,69 @@ def schedule(x):
 
         self._cooperator = Cooperator(scheduler=schedule)
 
+    @defer.inlineCallbacks
+    def _send_request_with_optional_trailing_slash(
+        self,
+        request,
+        try_trailing_slash_on_400=False,
+        backoff_on_404=False,
+        send_request_args={},
```

Comment:
I'd suggest making this a `**kwargs` so that you don't need to build a dict in the calling code.